include/boost/geometry/algorithms/detail/overlay/get_turn_info_for_endpoint.hpp

```diff
@@ -609,27 +578,23 @@ struct get_turn_info_for_endpoint
         *out++ = tp;
     }
 
-    template <typename SidePolicy>
-    static inline std::pair<operation_type, operation_type> operations_of_equal(SidePolicy const& side)
+    static inline std::pair<operation_type, operation_type>
+    operations_of_equal(int side_1, int side_2, int side_3)
     {
-        int const side_pk_q2 = side.pk_wrt_q2();
-        int const side_pk_p = side.pk_wrt_p1();
-        int const side_qk_p = side.qk_wrt_p1();
-
         // If pk is collinear with qj-qk, they continue collinearly.
         // This can be on either side of p1 (== q1), or collinear
         // The second condition checks if they do not continue
         // oppositely
-        if ( side_pk_q2 == 0 && side_pk_p == side_qk_p )
+        if ( side_1 == 0 && side_2 == side_3 )
```

Comment:
Out of curiosity, why did you decide to change the names of variables to less meaningful? Above I see a comment referencing specific points so this condition looks unrelated now.